src/Npgsql/PostgresDatabaseInfo.cs

```diff
@@ -202,12 +202,11 @@ internal async Task<List<PostgresType>> LoadBackendTypes(NpgsqlConnection conn,
 
                         var ns = reader.GetString(reader.GetOrdinal("nspname"));
                         var internalName = reader.GetString(reader.GetOrdinal("typname"));
-                        var oid = Convert.ToUInt32(reader[reader.GetOrdinal("oid")]);
+                        var oid = Convert.ToUInt32(reader.GetInt32(reader.GetOrdinal("oid")));
```

Comment:
Is this actually working? The reason that column access is strange, is that this is the initial type loading query, which means that types haven't been loaded yet :) Because of this chicken-and-egg problem, we set AllResultTypesAreUnknown on the comment, and everything just comes back as text, which we parse ourselves.

So I think the changes you have done will not work.

Reply:
I thought about moving this to a different PR since not related and will check.

Reply:
Ah so... Anyway this change is unrelated to this PR so will revert.

Reply:
OK. Definitely better to concentrate on finishing this PR and not putting anything extra inside.